File: expenses-manager-backend-java/src/main/java/com/github/beliakou/expenses/domain/Expense.java
package com.github.beliakou.expenses.domain;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;
import java.time.LocalDate;

@Document
public record Expense(@Id String id, String userId, LocalDate date, Category category, BigDecimal amount) {
}

File: expenses-manager-backend/expenses-manager-server/src/main/kotlin/com/github/beliakou/expenses/server/model/Expense.kt
package com.github.beliakou.expenses.server.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.math.BigDecimal
import java.time.LocalDate

@Document
data class Expense(@Id val id: String?,
                   val userId: String,
                   val date: LocalDate,
                   val category: String,
                   val amount: BigDecimal)